feature/consumer/retailer-list

Kwang-Kian-Hui opened the pull request:
Display retailer list on consumer home page.  Only feature, UI not complete.

Kwang-Kian-Hui committed:
Product object and Provider basic structure

Kwang-Kian-Hui committed:
Merge branch 'main' into origin/provider/models-and-riverpod-prov

Kwang-Kian-Hui committed:
Models and Providers

class and provider files for Product and Retailer

Kwang-Kian-Hui committed:
Widgets and some changes

Added widgets structure (listview, etc) for the consumer home page.
Remove ProviderScope() wrap of the AppWidget (ProviderScope() to wrap specific widgets that requires it).
Added 4 dummy data in retailerlist.

Kwang-Kian-Hui committed:
Merge branch 'main' into origin/provider/models-and-riverpod-prov

Kwang-Kian-Hui committed:
Merge branch 'main' into origin/provider/models-and-riverpod-prov

Kwang-Kian-Hui committed:
minor changes

merge from main branch and some changes

Kwang-Kian-Hui committed:
listview display dummy data

listview builder successfully displays dummy data. Added 1 additional dummy data. Basic listview item structure.

Kwang-Kian-Hui committed:
firebase test data

future builder and list view to retrieve firebase retailer test data

sinhaugoh commented on analysis_options.yaml:
```diff
@@ -1,6 +1,9 @@
 analyzer:
+  exclude:
+    - "**/*.g.dart"
+    - "**/*.freezed.dart"
```
what is the purpose of doing this?

Kwang-Kian-Hui replied:
Used to resolve an error previously but is now redundant, removed in later commits.

sinhaugoh commented on lib/consumer/home/presentation/drawer_widget.dart:
```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/material.dart';
```
this file should be under core/presentation as it is used across pages of some features

Kwang-Kian-Hui replied:
okay moved it over

sinhaugoh commented on lib/consumer/home/presentation/search_bar_filter.dart:
```diff
@@ -0,0 +1,13 @@
+import 'package:flutter/material.dart';
```
this should be a widget? can put inside a widget folder

sinhaugoh commented on lib/consumer/home/presentation/retailer_listview.dart:
```diff
@@ -0,0 +1,36 @@
+            } else {
+              return ListView.builder(
+                itemCount: snapshot.data!.length,
+                itemBuilder: (context, index) => ProviderScope(
```
i think the ProviderScope is redundant?

Kwang-Kian-Hui replied:
yeap you are right, removed in later commits

sinhaugoh reviewed (changes requested):
You can consider these changes :)

Kwang-Kian-Hui committed:
changes after feedback

Retailer list - remote service and repo. Possible failures of firestore.
Retailer data transfer objects(dto)

Kwang-Kian-Hui committed:
changes after feedback 2

Removed unnecessary lines of code in analysis_options.yaml.
Included retailer list state and notifier
Removal of no connection failure as error is not stated in the firebase doc
Changes made to error code to follow android error codes (previously was web error codes).

Kwang-Kian-Hui committed:
changes after feedback 3

notifier, provider files

Changed consumer home page of ConsumerWidget to ConsumerStatefulWidget.

get retailer list from firestore on initialisation of consumer home page.

Kwang-Kian-Hui commented:
Changes from feedback 3: also added query of retrieving retailers only if visibility is on

Kwang-Kian-Hui committed:
removed all product related files

all product related files will be in the feature/retailer/product-list branch

sinhaugoh commented on lib/consumer/application/retailer_list_notifier.dart:
```diff
@@ -0,0 +1,30 @@
+        orElse: () {
+          state = const RetailerListState.failure("Unknown error.");
+        },
+      );
```
seems like unknown and orelse return the same thing, maybe u can remove the unknown part?

Kwang-Kian-Hui replied:
okay, updated it on the latest commit

Kwang-Kian-Hui committed:
Update retailer list notifier

unnecessary unknown state setting removed as per requested changes.

Kwang-Kian-Hui committed:
Merge branch 'main' into feature/consumer/retailer-list
